app/src/main/java/erevacation/com/userlistanko/homescreen/list/ListRecyclerAdapter.kt

```diff
@@ -1,46 +1,66 @@
 import android.view.ViewGroup
 import android.widget.ImageView
 import android.widget.TextView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
 import erevacation.com.userlistanko.R
+import erevacation.com.userlistanko.datamodel.ListDM
 import org.jetbrains.anko.AnkoContext
 import org.jetbrains.anko.find
 
 class ListRecyclerAdapter(val presenter: ListFragmentPresenter) : RecyclerView.Adapter<ListRecyclerAdapter.ListViewHolder>() {
 
-    private val listInfo: MutableList<String> = mutableListOf("one", "two", "three", "four", "five")
-    @SuppressLint("PrivateResource")
-    private val listImage: MutableList<Int> = mutableListOf(R.drawable.ic_account_circle_black_24dp,
-            R.drawable.notification_bg_low_normal,
-            R.drawable.abc_ic_star_black_36dp,
-            R.drawable.ic_account_circle_black_24dp,
-            R.drawable.abc_ic_star_black_36dp)
+    private val listInfo: MutableList<ListDM> = mutableListOf()
+    private val profileInfoList: ArrayList<String> = arrayListOf()
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListViewHolder? {
         val itemView = AnkoViewHolderListUI().createView(AnkoContext.create(parent.context, parent))
         return ListViewHolder(itemView)
     }
 
     override fun onBindViewHolder(holder: ListViewHolder, position: Int) {
-        holder.bind(listInfo[position], listImage[position])
+        holder.bind(listInfo[position].name,
+                listInfo[position].surname,
+                listInfo[position].image)
+
+        holder.listCard.setOnClickListener {
+            this.profileInfoList.clear()
+            this.profileInfoList.add(listInfo[position].contactDetails.email)
+            this.profileInfoList.add(listInfo[position].contactDetails.address)
+            this.profileInfoList.add(listInfo[position].contactDetails.phones.home)
+            this.profileInfoList.add(listInfo[position].contactDetails.phones.work)
+
+            presenter.onContactClick(listInfo[position].name
+                    ,listInfo[position].surname
+                    ,listInfo[position].image
+                    ,profileInfoList)
+        }
     }
 
     override fun getItemCount(): Int {
         return listInfo.size
     }
 
-    class ListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-
-        var profileText: TextView = itemView.find(R.id.profile_text_viewholder)
-        val profileImage: ImageView = itemView.find(R.id.profile_circle_image_viewholder)
+    fun updateList(contacts: MutableList<ListDM>) {
+        this.listInfo.clear()
+        this.listInfo.addAll(contacts)
+        notifyDataSetChanged()
+    }
 
-        fun bind(list: String, image: Int) {
-            profileText.text = list
-            profileImage.setImageResource(image)
+    class ListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        var listCard: ConstraintLayout = itemView.find(R.id.list_card)
```

Comment:
can we make listCard private ?

Reply:
than we cant access it in OnBindViewHolder method